python/jupyterlab-chat/jupyterlab_chat/ychat.py

```diff
@@ -132,9 +133,19 @@ def add_message(self, new_message: NewMessage) -> str:
         message = Message(
             **asdict(new_message),
             time=timestamp,
-            id=uid
+            id=uid,
         )
 
+        # find all mentioned users and add them as message mentions
+        mention_pattern = re.compile("@([\w-]+):?")
+        mentioned_names: Set[str] = set(re.findall(mention_pattern, message.body))
+        users = self.get_users()
+        mentioned_usernames = []
```

Comment:
sanity-check: is there a risk that this list goes > 10 and that `len(users.items())` goes > 10?

If so, consider using a set here.

Reply:
Also good to check. This should be safe because we are iterating through the `users` dictionary to get all users, and checking to see if their mention name was in the set of mentions.

So since we check against the *set* of mentions, it doesn't matter if a user was mentioned 1 or 100 times; they will only be added to the `mentioned_usernames` list once. 👍